apps/extension/src/core/managers/transaction.ts

```diff
@@ -41,28 +52,49 @@ class TransactionManager extends BaseManager<Transaction> {
   init<TInput extends Input>(
     input: TInput,
     origin: OriginData,
-    options: CompletionOptions<ModelID | string, TInput>
+    options: CompletionOptions<ModelID | string, TInput> | MediaOptions<ModelID>
   ): Transaction {
     this._validateInput(input)
+  
+    // Extracting parameters common to both CompletionOptions and MediaOptions
     const {
-      temperature,
-      maxTokens,
-      stopSequences,
       model,
-      numOutputs = 1
+      numOutputs = 1,
     } = options
+  
+    // Extracting parameters specific to CompletionOptions
+    const {
+      temperature,
+      maxTokens,
+      stopSequences
+    } = options as CompletionOptions<ModelID | string, TInput>
```

Comment:
this is a little bit hacky, due to `as` here. I think it's better to say e.g. `"mimeType" in options ? ...` to avoid obscuring things from typescript

Reply:
refactored!